hack/generate-proto.sh

```diff
@@ -7,52 +7,51 @@ set -o pipefail
 source $(dirname $0)/library.sh
 header "generating proto files"
 
-ensure_protobuf
 ensure_vendor
 
-if [ "`command -v protoc-gen-gogo`" = "" ]; then
-  go install -mod=vendor ./vendor/github.com/gogo/protobuf/protoc-gen-gogo
-fi
-
-if [ "`command -v protoc-gen-gogofast`" = "" ]; then
-  go install -mod=vendor ./vendor/github.com/gogo/protobuf/protoc-gen-gogofast
-fi
-
-if [ "`command -v gogoproto`" = "" ]; then
-  go install -mod=vendor ./vendor/github.com/gogo/protobuf/gogoproto
-fi
-
-if [ "`command -v protoc-gen-go`" = "" ]; then
-  go install -mod=vendor ./vendor/google.golang.org/protobuf/cmd/protoc-gen-go
-fi
-
-if [ "`command -v protoc-gen-go-grpc`" = "" ]; then
-  go install -mod=vendor ./vendor/google.golang.org/grpc/cmd/protoc-gen-go-grpc
-fi
-
-if [ "`command -v protoc-gen-grpc-gateway`" = "" ]; then
-  go install -mod=vendor ./vendor/github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-grpc-gateway
-fi
-
-if [ "`command -v protoc-gen-openapiv2`" = "" ]; then
-  go install -mod=vendor ./vendor/github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2
-fi
-
-if [ "`command -v goimports`" = "" ]; then
-  go install -mod=vendor ./vendor/golang.org/x/tools/cmd/goimports
-fi
-
-export PATH="$(go env GOPATH)/bin:${PATH}"
-
 make_fake_paths
 export GOPATH="${FAKE_GOPATH}"
+export PATH="${GOPATH}/bin:${PATH}"
 cd "${FAKE_REPOPATH}"
 
+install-protobuf() {
+  # protobuf version
+  PROTOBUF_VERSION=27.2
+  PB_REL="https://github.com/protocolbuffers/protobuf/releases"
+  OS=$(uname_os)
+  ARCH=$(uname_arch)
+
+  echo "OS: $OS  ARCH: $ARCH"
+  BINARY_URL=$PB_REL/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-${OS}-${ARCH}.zip
+  if [[ "$OS" = "darwin" ]]; then
+    BINARY_URL=$PB_REL/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-osx-universal_binary.zip
+  elif [[ "$OS" = "linux" ]]; then
+    BINARY_URL=$PB_REL/download/v${PROTOBUF_VERSION}/protoc-${PROTOBUF_VERSION}-linux-x86_64.zip
+  fi
+  echo "Downloading $BINARY_URL"
+
+  tmp=$(mktemp -d)
+  trap 'rm -rf ${tmp}' EXIT
+
+  curl -sL -o ${tmp}/protoc-${PROTOBUF_VERSION}-${OS}-${ARCH}.zip $BINARY_URL
+  unzip ${tmp}/protoc-${PROTOBUF_VERSION}-${OS}-${ARCH}.zip -d ${GOPATH}
+}
+
+install-protobuf
+
+go install -mod=vendor ./vendor/github.com/gogo/protobuf/protoc-gen-gogo
```

Comment:
Always install these binaries to the temporary GOPATH to avoid version mismatch.

Reply:
This might slow down a little bit during local run (not too much), will be no difference for CI.